### PR TITLE
Cross-compile rule for 2.11

### PR DESCRIPTION
**Pull request by gabro**
This PR adds cross-compilation for 2.11, mostly to unblock https://github.com/scalameta/metals/issues/2099 (using the rule from Metals in 2.11 projects).

Unfortunately unused warnings reported by the 2.11 compiler are not as good as in 2.12+ so it's not  possible to accurately remove some imports in 2.11. This means I had to "fork" most tests relative to unused imports removal. I think it's an ok compromise in order to add 2.11 support.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -53,7 +53,16 @@ lazy val input = project
 
 lazy val output = project
   .dependsOn(shared)
-  .settings(skip in publish := true)
+  .settings(
+    skip in publish := true,
+    Compile / unmanagedSourceDirectories ++= {
+      val baseMainDir = baseDirectory.value / "src" / "main"
+      if (scalaVersion.value.startsWith("2.12.") || scalaVersion.value.startsWith("2.13."))
+        Seq(baseMainDir / "scala-2.12_2.13")
```
Comment (gabro) [inline build.sbt:61]: this is to avoid duplicating test outputs across 2.12 and 2.13.

**Comment by gabro** [comment]
Disclaimer:  I remember there were some issues with 2.11 and unused import warnings (especially regarding positions of importees), but I haven't spent a lot of time trying to work around them. It might be possible, but I don't have the time to try right now and I think adding "almost complete" 2.11 support is already a good improvement.

**Review by liancheng — APPROVED** [review]
LGTM, thanks!

**Comment by gabro** [comment]
@liancheng thanks for the quick review and merge, do you think it's possible to retroactively publish a 0.4.1 version for 2.11? Otherwise it's fine to move on with a new version and we'll update the dependency in Metals

**Comment by gabro** [comment]
Ah, I may have found something obvious I didn't consider for 2.11. The option to enable has a different name (`-Ywarn-unused-import`) and with that on we actually get something better. I'll send another PR soon

**Comment by liancheng** [comment]
@gabro, v0.4.1 has already been published a few days ago for fixing #101. I can publish v0.4.2 to include this one and #114. Does that sound good to you?

**Comment by liancheng** [comment]
Oh, I see, you mean publishing v0.4.1 just for Scala 2.11.

Hmm, TBH, I'm not quite sure how to do that. Thanks to @olafurpg's excellent sbt-ci-release plugin, I'm basically just pushing a Git tag to get all release work done. If you can provide detailed instructions about how to publish a Scala 2.11 only v0.4.1, I'm happy to follow that and give it a try. Otherwise, publishing a v0.4.2 is easier.

**Comment by gabro** [comment]
No problem at all, I also usually release from the CI!
Let's go ahead with 0.4.2 then, I'll follow up with and update in Metals

**Comment by olafurpg** [comment]
Backpublishing is not support by ci-release, I think it’s best to publish a new version instead

**Comment by liancheng** [comment]
OK, pushed the v0.4.2 tag, the release should be ready soon!

**Comment by gabro** [comment]
Thanks! 🙏

**Pull request merged**